src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -3310,6 +3361,17 @@ static void updateEntity
     toUnset.append(ENT_LOCATION, 1);
   }
 
+  // We don't touch toSet in the replace case, due to
+  // the way in which BSON is composed in that case (see below)
+  if (currentDateExpiration && (strcasecmp(action.c_str(), "replace") != 0))
```

Comment:
actionTypes changed recently. We are now using enumeration instead of strings. This would need adaptation after merging with master (have a look to https://github.com/telefonicaid/fiware-orion/pull/3197, the modification is quite straighforward).

Reply:
Fixed in faea551